### PR TITLE
XW-715 | Fix unit tests for ES6

### PR DESCRIPTION
**Pull request by hakubo**
https://wikia-inc.atlassian.net/browse/XW-715

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,8 +37,9 @@
     "wreck": "6.3.0"
   },
   "devDependencies": {
-    "babel-plugin-transform-es2015-modules-umd": "^6.0.15",
-    "babel-preset-es2015": "^6.1.4",
+    "babel-plugin-transform-es2015-modules-amd": "^6.1.18",
+    "babel-plugin-transform-es2015-modules-umd": "^6.1.18",
```
Comment (bognix) [inline package.json]: do we use UMD anywhere?

File: gulp/tasks/build-test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+gulp.task('build-test', function () {
+	return gulp.src([
+			path.join(paths.src, paths.jsFiles)
+		])
+		// @todo Fix in https://wikia-inc.atlassian.net/browse/XW-562
+		// .pipe(newer(path.join(paths.dest, folder + '.js')))
+		.pipe(babel({
+			presets: ['es2015'],
+			plugins: ['transform-es2015-modules-amd'],
```
Comment (bognix) [inline gulp/tasks/build-test.js]: we don't need any module loader to manage dependancies?

File: test/karma.conf.js [hunk 3]
```diff
@@ -8,10 +10,39 @@ module.exports = function (config) {
...
 		autoWatch: true,
 		singleRun: true,
-		browsers: ['PhantomJS'],
+		browsers: ['Chrome'],
 		// coverage reporter generates the coverage
 		reporters: ['progress', 'coverage'],
 
+		basePath: '../',
+		files: [
+			paths.vendor.dest + '/loader.js/loader.js',
```
Comment (bognix) [inline test/karma.conf.js]: can we add loader.js to bower.json?

Comment (bognix) [inline test/karma.conf.js]: done

File: front/scripts/main/models/discussion-index.js [hunk 4]
```diff
@@ -1,7 +1,6 @@
 import App from '../app';
 
-App.DiscussionIndexModel = Ember.Object.extend({
```
Comment (rogatty) [inline front/scripts/main/models/discussion-index.js]: It's not defined anywhere else so it should probably stay here.

File: gulp/tasks/build-test.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+	concat = require('gulp-concat'),
+	paths = require('../paths').scripts.front,
+	path = require('path');
+
+gulp.task('build-test', ['scripts-front'], function () {
+	return gulp.src([
+			path.join(paths.src, paths.jsFiles),
+			// The rest is built in scripts-front task
+			// FIXME uncomment when we know how to set Babel to generate proper module paths
+			// FIXME with the code above it's "auth/main", with the code below it's "main"
```
Comment (rogatty) [inline gulp/tasks/build-test.js]: Just a reminder that fixing this would speed up the test build.

Comment (rogatty) [inline gulp/tasks/build-test.js]: https://github.com/babel/babel/blob/8cd68ddc748ee7ca9fc7790673b9bc2df458f411/packages/babel-core/src/transformation/file/index.js#L184

**Pull request merged**